authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt

```diff
@@ -0,0 +1,189 @@
+ * tests.
+ * @param processLifecycleOwner Lifecycle owner that is used to listen for start/stop
+ * lifecycle events.
+ */
+class AuthenticatorBridgeManagerImpl(
+    private val applicationContext: Context,
+    private val connectionType: AuthenticatorBridgeConnectionType,
+    private val symmetricKeyStorageProvider: SymmetricKeyStorageProvider,
+    callbackProvider: AuthenticatorBridgeCallbackProvider = StubAuthenticatorBridgeCallbackProvider(),
+    processLifecycleOwner: LifecycleOwner = ProcessLifecycleOwner.get(),
+) : AuthenticatorBridgeManager, DefaultLifecycleObserver {
+
+    /**
+     * Keep track of whether the service is bound. Useful for making sure we don't call
+     * [Context.unbindService] on a service that isn't bound.
+     */
+    private var isBound = false
+
+    /**
+     * Main AuthenticatorBridgeService access point.
+     */
+    private var bridgeService: IAuthenticatorBridgeService? = null
+
+    /**
+     * Internal state of [accountSyncStateFlow].
+     */
+    private
+    val mutableSharedAccountsStateFlow =
+        MutableStateFlow<AccountSyncState>(AccountSyncState.Loading)
+
+    /**
+     * Callback registered with AuthenticatorBridgeService.
+     */
+    private val authenticatorBridgeCallback = callbackProvider.getCallback(::onAccountsSync)
+
+    /**
+     * Service connection that listens for connected and disconnected service events.
+     */
+    private val bridgeServiceConnection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName, service: IBinder) {
+            bridgeService = IAuthenticatorBridgeService.Stub.asInterface(service)
+
+            // TODO: Add check for version mismatch between client and server SDKs:
+            // TODO: https://livefront.atlassian.net/browse/BITAU-72
+
+            // Ensure we are using the correct symmetric key:
+            val localKeyFingerprint =
+                symmetricKeyStorageProvider.symmetricKey?.toFingerprint()?.getOrNull()
+
+            // Query bridge service to see if we have a matching symmetric key:
+            val haveCorrectKey = bridgeService
+                .safeCall { checkSymmetricEncryptionKeyFingerprint(localKeyFingerprint) }
+                ?: false
+
+            if (!haveCorrectKey) {
+                // If we don't have the correct key, query for key:
+                symmetricKeyStorageProvider.symmetricKey =
+                    bridgeService.safeCall { symmetricEncryptionKeyData }
+            }
+
+            if (symmetricKeyStorageProvider.symmetricKey == null) {
+                // This means bridgeService returned a null key, which means we can make
+                // no valid operations. We should disconnect form the service and expose to the
+                // calling application that authenticator sync is not enabled.
+                mutableSharedAccountsStateFlow.value = AccountSyncState.SyncNotEnabled
+                unBindService()
```

Comment:
Are you missing a `return` here?

Reply:
Ah, yep!

This was working before because `unBindService` prevents anything else from happening. But the early return makes it more explicit. 9f8aed67773b46ed30400943194f9f3cab0a9c81